### ISSUE TITLE
Not mock properly for methods with optional params

### ISSUE THREAD

**Issue by tommy20061222** [matthewmcnew/safe-mock#20]
It throws sytax error when we use verify...calledWith on the methods with optional params.

**Comment by matthewmcnew** [comment]
Hi Tommy, I will look into it! Thanks for the heads up!

**Comment by matthewmcnew** [comment]
@tommy20061222 This appears to be an issue with how Typescript handles generics within safe-mock. So, I added the ability to use verify in unsafe mode to get around this. 

```typescript
import SafeMock, {verify} from "safe-mock";

interface SomeService {
    someMethod(argument?: number, otherArg?: string): string;
}
const mock = SafeMock.build<SomeService>();

//use verify.unsafe when working with optional parameters
//unfortunately, type safety is lost when using 'verify.unsafe'  
verify.unsafe(mock.someMethod).calledWith(123, "someArg");
verify.unsafe(mock.someMethod).calledWith(123);
verify.unsafe(mock.someMethod).called();
```

This is available in 0.2.11
